static/js/shared.js

```diff
@@ -140,7 +140,30 @@ window.onerror = function(error, url, line) {
     }
 };
 
-function warn(msg){
+function warn(errors){
+    function formatErrors(errors){
+        var table = document.createElement("table");
+        table.className = "warn-table"
+        for(var id in errors){
+            var tr = document.createElement('tr');
+            var td1 = document.createElement('td');
+            var td2 = document.createElement('td');
+            var text1 = document.createTextNode(errors[id]['name']);
+            var text2 = document.createTextNode(errors[id]['msg']);
+            td1.appendChild(text1);
+            td2.appendChild(text2);
+            tr.appendChild(td1);
+            tr.appendChild(td2);
+            table.appendChild(tr);
+        }
+        $("#alert-text").after(table)
+    }
     document.getElementById("alert-modal").style.display="block";
```

Comment:
should we re-use the alert-modal - OR - should we create a new global modal box specific for alerts? (we have 2 right now, alert-modal and duk-modal, which should really be consolidated as the only difference is an image)

Reply:
Hmm... I kinda wished there was a new/different one when working on this as I tried to do everything without messing the styling for the existing uses.  So if this has a specific use case and graphical requirements it would make sense to make it a new modal, or even a dynamic modal (created entirely in JS and added to the DOM).